### PR TITLE
Epic/9/issue/352/groupless user list

### LINKED ISSUE #352: [Backend] Endpoint to return a list of users with STT and Region
Description:
This issue will deliver a `GET` endpoint of users with no role assigned. 

### Tasks
- [ ] Tests to cover list of users with STT and Region endpoint
- [ ] Add STT and Region results to User Endpoint

### AC
- [ ] Demonstrated `GET` request will return results like the following
```
{
  "id": 1,
  "first_name": "John",
  "last_name": "Doe",
  "stt": {
    "id": 1,
    "type": "state",
    "code": "AL",
    "name": "Alabama"
  },
  "region": 1
},
{
  "id": 2,
  "first_name": "Jane",
  "last_name": "Doe",
  "stt": {
    "id": 15,
    "type": "state",
    "code": "MI",
    "name": "Michigan"
  },
  "region": 3
}
```
### Notes
- An endpoint already exists for a list of users, so the task here is merely to add STT and region data to the results

### PR DESCRIPTION
**Pull request by riatzukiza**
### This pull request changes...

Adds an endpoint to list all users with out a group
Closes #352 

## FUNCTIONALITY
*Describe the functionality*
*Addresses Issue #352*

## TO TEST

*List the steps to test the PR*

### This pull request is ready to merge when...

- [ ] Meets acceptance criteria for the issue 
- [ ] Code is reviewed by someone other than the original author
- [ ] Code is tested 
- [ ] Experience is approved by UX  
- [ ] Meets [a11y checklist](https://github.com/HHS/TANF-app/blob/main/docs/a11y/how-18f-will-test-a11y.md)
- [ ] Meets [Raft's Manual QASP Checklist](https://teams.microsoft.com/l/file/B5285301-13CA-46FD-A723-AFFD578C224A?tenantId=d58addea-5053-4a80-8499-ba4d944910df&fileType=docx&objectUrl=https%3A%2F%2Fhhsgov.sharepoint.com%2Fsites%2FTANFDataPortalOFA%2FShared%20Documents%2FGeneral%2FManual%20Testing%20Practices_Issue173.docx&baseUrl=https%3A%2F%2Fhhsgov.sharepoint.com%2Fsites%2FTANFDataPortalOFA&serviceName=teams&threadId=19:f769bbcb029f4f02b55ae7fad90e310d@thread.skype&groupId=41f194a6-c1d3-4680-933e-c8ee7d17e287):lock: (only applicable for PR to main HHS)
- [ ] Documentation is updated 
  - OpenAPI  
  - Readme
  - Entity Relationship Diagram (ERD)

### CODE REVIEW

**Review by carltonsmith — CHANGES_REQUESTED** [review]
@riatzukiza This looks great, but there are some linting problems. If you can update those we can continue to Raft Review

File: tdrs-backend/docs/api/groupless_users.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    "first_name":"Jane",
+    "last_name":"Doe"
+  },
+  {
+    "id":"b0a5525c-e6e5-ff9a-8eb6-3fffaf3fb0a5a6b3",
+    "username":"user@example.com",
+    "first_name":"Jane",
+    "last_name":"Doe"
+  }
+]
```
Comment (carltonsmith) [inline tdrs-backend/docs/api/groupless_users.md:43]: I think this list should include the STT as the OFA Analysts and Admins will need to know that to help identify the user for approval and role assignment

File: tdrs-backend/docs/api/groupless_users.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+Parameters:
+
+
+ A valid httpOnly cookie in the request header to track the users session
+
+*Note:*
+
+- Authorization Protected 
+
```
Comment (carltonsmith) [inline tdrs-backend/docs/api/groupless_users.md:17]: This list should only be accessible to the OFA Admin and Superuser roles (and maybe OFA Analyst, I will confirm). We should have that in the documentation.

File: tdrs-backend/docs/api/groupless_users.md [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+```
+----
+**Calls made by authorized users who aren't Admins:**
+```json
+Content-Type application/json
+500 Internal Server Error
+
+System Error Message:
+Does Not Exist
+Group matching query does not exist.
```
Comment (carltonsmith) [inline tdrs-backend/docs/api/groupless_users.md:75]: Shouldn't this return a `403 FORBIDDEN` response?

File: tdrs-backend/tdpservice/users/test/test_api.py [hunk 4]
```diff
@@ -526,6 +526,27 @@ def test_permission_delete(api_client, create_test_users):
...
+    response_users = list(map(lambda u: u.get("id"), response.data))
+
+    assert all(
+        map(
+            lambda groupless_id, response_id: groupless_id == response_id,
+            groupless_users,
+            response_users,
+        )
+    )
+
```
Comment (carltonsmith) [inline tdrs-backend/tdpservice/users/test/test_api.py:548]: I think a test confirming that a user with a group does not appear in the list would be helpful

**Review by carltonsmith — CHANGES_REQUESTED** [review]
Great work! There are a couple of adjustments, but not much.

**Comment by jtwillis92** [comment]
@riatzukiza @carltonsmith is this PR still in progress or has this work been postponed?

**Comment by carltonsmith** [comment]
> @riatzukiza @carltonsmith is this PR still in progress or has this work been postponed?

Yes, this changed when we decided to go with Django Admin. I'll close it now.

**Pull request closed without merging**